tests/e2e/core-tests/specs/shopper/front-end-checkout-coupons.test.js

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable jest/no-export, jest/no-disabled-tests, jest/expect-expect, jest/no-standalone-expect */
+/**
+ * Internal dependencies
+ */
+const {
+	shopper,
+	merchant,
+    createCoupon,
+	createSimpleProduct,
+	uiUnblocked
+} = require( '@woocommerce/e2e-utils' );
+
+/**
+ * External dependencies
+ */
+const {
+	it,
+	describe,
+	beforeAll,
+} = require( '@jest/globals' );
+
+const runCheckoutApplyCouponsTest = () => {
+	describe('Checkout applying coupons', () => {
+		beforeAll(async () => {
+			await merchant.login();
+            await createSimpleProduct();
+            await createCoupon('Fixed cart discount');
```

Comment:
You can store the dynamic generated coupon code returned by `createCoupon` by creating a variable outside the `beforeAll`. eg

```
let couponFixedCart;
beforeAll(...
// ...
couponFixedCart = await createCoupon(5, 'Fixed cart discount');
```

Reply:
Good idea! I implemented it that way now.